src/components/TopicViews.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import PreviewCard from './PreviewCard';
+import { RootState } from '../store/reducers';
+import { ResourcesSlice } from '../store/resourcesSlice';
+import TopicHeader from './TopicHeader';
+
+interface TopicViewProps {
+  topicId: number, 
+}
+
+function TopicViews( { topicId }: TopicViewProps) {
+  const allTopics = useSelector((state: RootState) => state.topics); 
+  const currentTopic = allTopics[topicId]; 
+  const name = currentTopic.name;
+
+  const selectTopicResources = (state: RootState) =>
+    Object.keys(state.resources as ResourcesSlice)
+      .filter(
+        (id) =>
+          state.resources[(id as unknown) as number].tags.includes(name, 0) &&
```

Comment:
You can actually replace all of the `(id as unknown) as number` with just `id`! Earlier I was doing this to try to cast id as a number type, but apparently all keys in Javascript are strings no matter what, so just using `id` by itself (which is a string) works fine

Reply:
Same with `(key as unknown) as number` => `key`